### PR TITLE
OOP school library: decorate a class Feature/project 2 

### PR DESCRIPTION
**Pull request by david-lafontant**
### OOP school library: decorate a class

- Implement a decorator design pattern to correct the names of people.
- Implement [Project requirements](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_decorator_v2.md#:~:text=Project%20requirements,trim%20the%20word.)

### CODE REVIEW

File: src/decorator.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  end
+end
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    if @nameable.correct_name.length > 10
+      @nameable.correct_name.slice(0, 10)
+    else
+      @Nameable.correct_name
+    end
```
Comment (MrBrN197) [inline src/decorator.rb]: - [ ] Hello, Please kindly notice that you have a typo here. this is the incorrect variable name. you have defined the `@name`  instance variable which is what you should be using here. please make this change.

**Review by MrBrN197 — CHANGES_REQUESTED** [review]
## Hi, @david-lafontant  👋 

You've done a really nice job so far.
There are some issues that you still need to work on to go to the next project but you are almost there!
## To highlight 👏👏
- ✔ linter checks passing correctly

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/decorator.rb (hunk 1 above)
Comment (david-lafontant) [reply src/decorator.rb]: Apologies. Thank you very much @MrBrN197   .

**Review by Gambit142 — APPROVED** [review]
Hi @david-lafontant 👋,

Great work implementing the change suggested by the previous reviewer.

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by david-lafontant** [comment]
> Hi @david-lafontant 👋,
> 
> Great work implementing the change suggested by the previous reviewer.
> 
> Your project is complete! There is nothing else to say other than... it's time to merge it :shipit: Congratulations! 🎉
> 
> Cheers and Happy coding!👏👏👏
> 
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
> 
> _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._


Thank you @Gambit142 for the time taken to review this project.

**Pull request merged**